.github/workflows/inspektor-gadget.yml

```diff
@@ -210,6 +220,29 @@ jobs:
         # For the moment, we only support these two platforms.
         platforms: linux/amd64,linux/arm64
 
+    - name: Build gadget ${{ matrix.type }} container image as artifacts
+      if: startsWith(github.ref, 'refs/tags/v') != true && github.event_name != 'schedule'
+      uses: docker/build-push-action@v2
+      with:
+        context: /home/runner/work/inspektor-gadget/inspektor-gadget/
+        file: /home/runner/work/inspektor-gadget/inspektor-gadget/Dockerfiles/gadget-${{ matrix.type }}.Dockerfile
+        build-args: |
+          ENABLE_BTFGEN=true
+        outputs: type=docker,dest=/tmp/gadget-image-${{ matrix.type }}-linux-amd64.tar
+        tags: ${{ steps.choose-repo-determine-image-tag.outputs.container-repo }}:${{ steps.choose-repo-determine-image-tag.outputs.image-tag }}
+        cache-from: type=local,src=/tmp/.buildx-cache
+        cache-to: type=local,dest=/tmp/.buildx-cache-new
+        # docker exporter doesn't allow exporting multi-platform images
+        platforms: linux/amd64
```

Comment:
What about using the same logic to build the image for the different scenarios. Something like:

```
    - name: Build gadget ${{ matrix.type }} container image
      uses: docker/build-push-action@v2
      with:
        context: /home/runner/work/inspektor-gadget/inspektor-gadget/
        file: /home/runner/work/inspektor-gadget/inspektor-gadget/Dockerfiles/gadget-${{ matrix.type }}.Dockerfile
        build-args: |
          ENABLE_BTFGEN=true
        # Only build the image, we'll push or save as an artifact in another step
        push: false
        tags: ${{ steps.choose-repo-determine-image-tag.outputs.container-repo }}:${{ steps.choose-repo-determine-image-tag.outputs.image-tag }}
        cache-from: type=local,src=/tmp/.buildx-cache
        cache-to: type=local,dest=/tmp/.buildx-cache-new
        # For the moment, we only support these two platforms.
        platforms: linux/amd64,linux/arm64
    - name: push to container registry 
      if: startsWith(github.ref, 'refs/tags/v') || github.event_name == 'schedule'
      run: | 
        docker push ...
   - name: save image
     if: startsWith(github.ref, 'refs/tags/v') != true && github.event_name != 'schedule'
     run: | 
       docker save? 
   - name: upload artifact
    // some logic you currently have on the PR
 ```

Reply:
That sounds better and will resolve above confusion. As per [this](https://github.com/docker/buildx/issues/166) the current docker tools won't allow pushing individual images unless we use `docker manifest` to merge manifest list but let me have a look into this approach.

Reply:
Perhaps you can change the value of `push` of `docker/build-push-action` depending on github.ref: (I haven't tested it at all, just throwing an idea). 

```
 - name: Build gadget ${{ matrix.type }} container image
      uses: docker/build-push-action@v2
      with:
        context: /home/runner/work/inspektor-gadget/inspektor-gadget/
        file: /home/runner/work/inspektor-gadget/inspektor-gadget/Dockerfiles/gadget-${{ matrix.type }}.Dockerfile
        build-args: |
          ENABLE_BTFGEN=true
        # Only push when building a release or main branch
        push: startsWith(github.ref, 'refs/tags/v') || github.event_name == 'schedule'  # todo: update for main branch. 
        tags: ${{ steps.choose-repo-determine-image-tag.outputs.container-repo }}:${{ steps.choose-repo-determine-image-tag.outputs.image-tag }}
        cache-from: type=local,src=/tmp/.buildx-cache
        cache-to: type=local,dest=/tmp/.buildx-cache-new
        # For the moment, we only support these two platforms.
        platforms: linux/amd64,linux/arm64
```

Reply:
I [merged](https://github.com/kinvolk/inspektor-gadget/pull/871/files#diff-bda633e49a0d242b6842493c3896d4f766f361166fa4c08e1c2e01d827544cadR229) both steps now mainly because I want to make the tar image tested and the one pushed to registry must be same. The `push: true` or `load:true` doesn't allow that. Hopefully we will have [this](https://github.com/docker/buildx/issues/59#issuecomment-1189999088) soon and can make it cleaner further.